src/wasm/literal.cpp

```diff
@@ -2358,4 +2396,18 @@ Literal Literal::swizzleVec8x16(const Literal& other) const {
   return Literal(result);
 }
 
+bool Literal::isSubRtt(const Literal& other) const {
+  assert(type.isRtt() && other.type.isRtt());
+  if (type == other.type) {
+    return true;
+  }
+  // Look up the chain to see if the other RTT is one of our parents.
+  for (auto super : getRttSupers()) {
+    if (super == other.type) {
+      return true;
+    }
+  }
```

Comment:
Good question, you might be right. I don't understand the GC docs well enough I guess. It seems memory-safe to not care about the whole chain, but there is the aspect of making the check fast that I don't quite get (and maybe checking the whole chain does that?).

If you're sure then I can rewrite it to do that. Otherwise I can open an issue on the GC repo.

Reply:
I think looking at the whole chain make more sense because otherwise there's really no reason we maintain the chain, but I don't know GC much and I'm not sure. If you can open an issue that'd be good. 

Also I'm not sure why checking the whole chain may not be memory-safe.

Reply:
I opened https://github.com/WebAssembly/gc/issues/170

You are probably right. But the chain is still crucial in my understanding - it ensures memory safety. If we have an object with RTT x, and the chain has types a,b,c,d, then we know from that chain that it is memory safe to cast that object to any of those types.

About your last comment - sorry, I wasn't clear I guess. I agree checking the whole chain is memory safe - it adds more checks on top of my understanding, without losing any.